client/src/docs-ui/ui-component-props/ui-component-props.tsx

```diff
@@ -18,66 +27,59 @@ export class DocsUIComponentProps {
+      </docs-in-page-link>
+    ) : (
+      <h4>Properties</h4>
+    );
+  }
 
+  content() {
+    if (this.propType === "attr") {
+      return (
+        <p>
+          <code>{this.component?.tag}</code>
+          provides the following properties to configure the component.
+        </p>
+      );
+    } else if (this.propType === "css") {
+      return (
+        <p>
+          <code>{this.component?.tag}</code>
+          provides the following
+          <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties">
+            css properties
+          </a>
+          to modify the style at component level.
+        </p>
+      );
+    } else if (this.propType === "slots") {
+      return (
+        <p>
+          <code>{this.component?.tag}</code>
+          provides the following slots based off of the
+          <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot">
+            Web Components slot
+          </a>
+          element.
+        </p>
+      );
+    }
+  }
+
+  render() {
+    console.log(this.propType);
+    if (!this.component || !this.component.tag) return;
+    const tableGenerator = tableGeneratorMap[this.propType];
     return (
       <Host>
-        {this.useTableHeaders ? (
-          <docs-in-page-link targetId={sectionId}>
-            <h2 id={sectionId}>Properties</h2>
-          </docs-in-page-link>
-        ) : (
-          <h4>Properties</h4>
-        )}
-
-        {this.component?.props.map((prop) => {
-          const groupId = `prop-${prop.attr || String(count)}`;
-          if (!prop.attr) {
-            count++;
-          }
-
-          return (
-            <table class={tableStyle} key={groupId}>
-              <thead>
-                <tr>
-                  <th colSpan={2}>
-                    {this.useTableHeaders ? (
-                      <docs-in-page-link targetId={groupId}>
-                        <h3 id={groupId}>{prop.name}</h3>
-                      </docs-in-page-link>
-                    ) : (
-                      <div class={tableHeaderStyle}>{prop.name}</div>
-                    )}
-                  </th>
-                </tr>
-              </thead>
-              <tbody>
-                {prop.attr && (
-                  <tr>
-                    <th>Attribute</th>
-                    <td>{prop.attr}</td>
-                  </tr>
-                )}
-                <tr>
-                  <th>Description</th>
-                  <td>{prop.docs}</td>
-                </tr>
-                <tr>
-                  <th>Type</th>
-                  <td>{prop.type}</td>
-                </tr>
-                {prop.default && (
-                  <tr>
-                    <th>Default</th>
-                    <td>{prop.default}</td>
-                  </tr>
-                )}
-              </tbody>
-            </table>
-          );
-        })}
+        {this.header()}
+        {this.content()}
```

Comment:
Can we validate this doesn't cause node sharing? More info on Stencil's docs [here](https://stenciljs.com/docs/templating-jsx#avoid-shared-jsx-nodes)

Reply:
Yes! The two are completely separate nodes.